deploy/firesim

```diff
@@ -358,7 +358,7 @@ def shareagfi(build_config_file: BuildConfigFile) -> None:
     # share the image with all users
     for agfiname in build_config_file.agfistoshare:
         agfi = build_config_file.hwdb.get_runtimehwconfig_from_name(agfiname).agfi
-
+        assert agfi is not None
```

Comment:
Provide a message here

Reply:
Consider refactoring into a `no_longer_optional(agfi)` helper utility if mypy is smart enough.